tests/testthat/test-utils-plot.R

```diff
@@ -0,0 +1,358 @@
+      view(xlim = c(startDate, endDate2))
+
+    newPlot1 <- repgen:::XAxisLabelStyle(plot_object1, startDate, endDate1, timezone, plotDates1)
+    newPlot2 <- repgen:::XAxisLabelStyle(plot_object2, startDate, endDate2, timezone, plotDates2)
+
+    mtext1 <- gsplot:::views(newPlot1)[[1]][which(grepl("mtext", names(gsplot:::views(newPlot1)[[1]])))]
+    mtext2 <- gsplot:::views(newPlot2)[[1]][which(grepl("mtext", names(gsplot:::views(newPlot2)[[1]])))]
+
+    expect_equal(length(mtext1), 0)
+    expect_equal(length(mtext2), 2)
+})
+
+test_that('Calculate Limits properly calculates the limits of the provided data', {
+    timezone <- "Etc/GMT+5"
+
+    pts1 <- data.frame(
+        time = c(repgen:::flexibleTimeParse("2013-11-10T12:00:00-05:00", timezone), 
+                 repgen:::flexibleTimeParse("2013-12-02T23:59:00-05:00", timezone)),
+        value = c(10, 20)
+    )
+
+    pts2 <- data.frame(
+        time = c(repgen:::flexibleTimeParse("2013-11-10T12:00:00-05:00", timezone), 
+                 repgen:::flexibleTimeParse("2013-12-02T23:59:00-05:00", timezone),
+                 repgen:::flexibleTimeParse("2014-11-12T23:59:00-05:00", timezone)),
+        value = c(10, 20, 11)
+    )
+
+    lims1 <- repgen:::calculateLims(pts1)
+    lims2 <- repgen:::calculateLims(pts2)
+
+    expect_is(lims1, 'list')
+    expect_is(lims2, 'list')
+    expect_equal(lims1$xlim[[1]], pts1[['time']][[1]])
+    expect_equal(lims1$xlim[[2]], pts1[['time']][[2]])
+    expect_equal(lims1$ylim[[1]], 10)
+    expect_equal(lims1$ylim[[2]], 20)    
+    
+    expect_equal(lims2$xlim[[1]], pts2[['time']][[1]])
+    expect_equal(lims2$xlim[[2]], pts2[['time']][[3]])
+    expect_equal(lims2$ylim[[1]], 10)
+    expect_equal(lims2$ylim[[2]], 20)
+})
+
+test_that('plotItem properly adds an item to a GSPlot object with the proper styles', {
+    timezone <- "Etc/GMT+5"
+    startDate <- repgen:::flexibleTimeParse("2013-11-10T12:00:00-05:00", timezone)
+    endDate <- repgen:::flexibleTimeParse("2013-12-02T23:59:00-05:00", timezone)
+
+    plotDates <- c(
+         as.Date("2013-11-10T00:00:00-05:00"),
+         as.Date("2013-11-17T00:00:00-05:00"),
+         as.Date("2013-11-24T00:00:00-05:00"),
+         as.Date("2013-12-01T00:00:00-05:00")
+    )
+
+    pts <- data.frame(
+        time = c(repgen:::flexibleTimeParse("2013-11-11T12:00:00-05:00", timezone), 
+                 repgen:::flexibleTimeParse("2013-11-12T23:59:00-05:00", timezone)),
+        value = c(10, 20)
+    )
+
+    plot_object <- gsplot(ylog = FALSE, yaxs = 'i') %>%
+      grid(nx = 0, ny = NULL, equilogs = FALSE, lty = 3, col = "gray") %>%
+      axis(1, at = plotDates, labels = format(plotDates, "%b\n%d"), padj = 0.5) %>%
+      axis(2, reverse = FALSE, las=0) %>%
+      view(xlim = c(startDate, endDate))
+
+    plot_object <- repgen:::plotItem(plot_object, pts, repgen:::getDVHydrographPlotConfig, list(pts, 'groundWaterLevels'), isDV=TRUE)
+  
+    expect_equal(length(plot_object$view.1.2$points$x), 2)
```

Comment:
:+1: 

@lindsaycarr any reservations about us testing gsplot like this? EG: any plans of internal structures/implementations changing soon?